src/main/java/rx/internal/operators/OperatorMergeEmptyWith.java

```diff
@@ -0,0 +1,75 @@
+/**
+ * Copyright 2014 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package rx.internal.operators;
+
+import rx.Observable;
+import rx.Observable.Operator;
+import rx.Producer;
+import rx.Subscriber;
+
+/**
+ * Returns an Observable that emits an error if any item is emitted by the source and emits items from the supplied
+ * alternate {@code Observable}. The errors from source are propagated as-is.
+ *
+ * @param <T> the source value type
+ * @param <R> the result value type
+ */
+public final class OperatorMergeEmptyWith<T, R> implements Operator<R, T> {
+
+    private final Observable<? extends R> alternate;
+
+    public OperatorMergeEmptyWith(Observable<? extends R> alternate) {
+        this.alternate = alternate;
+    }
+
+    @Override
+    public Subscriber<? super T> call(Subscriber<? super R> child) {
+        final ParentSubscriber parent = new ParentSubscriber(child);
+        child.add(parent);
+        alternate.unsafeSubscribe(child);
+        return parent;
+    }
+
+    private final class ParentSubscriber extends Subscriber<T> {
+
+        private final Subscriber<? super R> child;
+
+        ParentSubscriber(Subscriber<? super R> child) {
+            this.child = child;
+        }
+
+        @Override
+        public void setProducer(final Producer producer) {
+            /*Nothing from this producer will ever be requested as we never expect any items to be emitted from the
+            parent.*/
+        }
+
+        @Override
+        public void onCompleted() {
+            // Nothing to do as the child completes with the alternate observable.
+        }
+
+        @Override
+        public void onError(Throwable e) {
+            child.onError(e);
```

Comment:
I made the `ChildSubscriber` as `SerializedSubscriber`. 
On thinking more about the `unsubscribe` part, I think I am not yet clear as to why that is a problem. AFAIU, on parent error, it will be propagated to the child as an `onError` after which the `SafeSubscriber` at the end of the chain should `unsubscribe`, which will flow in and unsubscribe from the parent and child. Is my understanding incorrect?